### PR TITLE
switch to lcg 106

### PR DESCRIPTION
**Pull request by harrypuuter**
This PR updates some CROWN dependencies

- [x] Switch to LCG 106 and GCC 13
- [ ] Switch to C++ 20 Standard
- [x] Switch to spdlog 1.14.1
- [x] Drop RHEL 8 Support

### CODE REVIEW

**Comment by harrypuuter** [comment]
fixes #264

File: include/utility/Logger.hxx [hunk 1]
```diff
@@ -1,12 +1,96 @@
...
+// Register the formatter with fmt library for any size of std::bitset
+template <std::size_t N>
+struct fmt::formatter<std::bitset<N>> : formatter<std::string> {
+    template <typename FormatContext>
+    auto format(const std::bitset<N> &b,
+                FormatContext &ctx) -> decltype(ctx.out()) {
+        return BitsetFormatter<N>::format(ctx, b);
+    }
+};
+
```
Comment (nshadskiy) [inline include/utility/Logger.hxx:93]: Does this already cover all the cases that we have in Crown?

Comment (harrypuuter) [reply include/utility/Logger.hxx:93]: most basic types are covered, only the ones that are not "common" like the ROOT special types. At least the tau analysis does compile with this. Maybe other analyses need additional types here

**Review by nshadskiy — APPROVED** [review]
Looks good to me!

**Pull request merged**